src/app/pages/news/ListView.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from 'react';
+import NewDetailView from './NewDetailView';
+import * as bulmaStyles from 'bulma';
+import { stat } from 'fs';
+import customNewsStyle from './style.scss';
+
+
+function ListView(props) {
+
+    const [isDetailOpen, setDetail] = useState(false);
+    const [clickedNew, setclickedNew] = useState(props.news[0]);
+    const [fromAllNews, getFromAllNews] = useState(false);
+    const [seeAllNews, setAllNews] = useState(false);
+    const [data, setData] = useState(props.news);
+    const [filterState, setFilterState] = useState(false);
+
+    var tags = [];
+    var newsList = [];
+    if (props.news.length > 0) newsList = props.news;
+    props.news.map((item, i) => {
+        item.tags.map((tag, i) => {
+            if (!tags.includes(tag)) tags.push(tag)
+        })
+    }); 
```

Comment:
burada bir karmaşa var, tag'e göre filtreleme yapıyorsak map methodunu tam anlamıyla kullanmamız gerekiyor. map aslında bir döngü kurmak için değil, belirli bir array çıktısını oluşturmak için kullanılıyor.

Dolayısıyla 18 satırdan bu yana olan kodu şöyle yazmalıydık:

```js
const newsList = props.news;
const tags = newsList.reduce(
  (acc, cur) => [ ...acc, ...cur.tags.filter(x => !acc.includes(x)) ],
  []
);
```